pkg/build/builder/cmd/builder.go

```diff
@@ -41,22 +41,28 @@ func run(builderFactory factoryFunc) {
 		authcfg     docker.AuthConfiguration
 		authPresent bool
 	)
+	output := true
 	if len(build.Parameters.Output.DockerImageReference) == 0 {
 		if build.Parameters.Output.To != nil {
 			log.Fatalf("Cannot determine an output image reference. Make sure a registry service is running.")
 		}
-		log.Fatal("Build output has an empty Docker image reference.")
+		output = false
 	}
-	registry, _, _, _, err := image.SplitDockerPullSpec(build.Parameters.Output.DockerImageReference)
-	if err != nil {
-		log.Fatalf("Build output does not have a valid Docker image reference: %v", err)
+	if output {
+		registry, _, _, _, err := image.SplitDockerPullSpec(build.Parameters.Output.DockerImageReference)
+		if err != nil {
+			log.Fatalf("Build output does not have a valid Docker image reference: %v", err)
+		}
+		authcfg, authPresent = dockercfg.NewHelper().GetDockerAuth(registry)
 	}
-	authcfg, authPresent = dockercfg.NewHelper().GetDockerAuth(registry)
-
 	b := builderFactory(client, endpoint, authcfg, authPresent, &build)
 	if err = b.Build(); err != nil {
```

Comment:
STI build will fail here, since not passing the name of the output image which is required (see [sti.go#34](https://github.com/openshift/origin/blob/master/pkg/build/builder/sti.go#L34)).